.github/workflows/ci.yml

```diff
@@ -943,6 +943,9 @@ jobs:
     env:
       # No dbus in GitHub Actions
       PYTHON_KEYRING_BACKEND: keyrings.alt.file.PlaintextKeyring
+    permissions:
+      # For trusted publishing
+      id-token: write
```

Comment:
This is dubious, no?

Reply:
It's from the docs:

![image](https://github.com/user-attachments/assets/ffbc48c7-5a17-4fcc-a14f-ec8e220b2abc)

https://docs.pypi.org/trusted-publishers/using-a-publisher/

Reply:
It seems like it still may be dubious to run in our public CI, right?

Reply:
We of course need to review all changes the code called here, but the risk is otherwise the same as if someone makes a PR that adds `id-token: write`. The permission in testpypi itself is clearly scoped to allow only this package and only this environment.